src/cargo/util/config.rs

```diff
@@ -545,6 +545,14 @@ impl Config {
         }
     }
 
+    /// Gets the index for a registry.
+    pub fn get_registry_index(&self, registry: &str) -> CargoResult<String> {
```

Comment:
Perhaps this could bake in returning a `Url` instead of a `String`?

Reply:
Sounds reasonable, I will update.